### ISSUE TITLE
The training and inference results are not consistent.

### ISSUE THREAD

**Issue by black-tea01** [2U1/Qwen2-VL-Finetune#67]
The training and inference results are not consistent. I observed the prediction results during training without any problem, but the inference found a significant difference from the training, which confused me. I analyzed and found that it's not the problem of not loading all the model weights, because when loading the trained model, the result of the first step is already correct, but the inference result is incorrect. Is there a direct code difference between training and inference? Are there any other possibilities that may have caused this problem?
cfg: zero3, video train

**Comment by 2U1** [comment]
How is it different?

**Comment by black-tea01** [comment]
The logits output during training are correct, however, using the same sample, the logits in testing do not match the training, for example, the options in training may be A or B. But in the test, the result is always B, even though I can lower the threshold by using logits scores to obtain a generally correct result.

**Comment by 2U1** [comment]
@black-tea01 I couldn't find the issue for in consistant answering. Sorry for the late response.
